.ci/Jenkinsfile

```
@@ -282,31 +288,90 @@ pipeline {
+        allOf {
+          branch 'master'
+          expression { return params.release }
+        }
       }
       stages {
-        stage('Opbeans') {
-          environment {
-            REPO_NAME = "${OPBEANS_REPO}"
+        stage('Release CI') {
+          options {
+            skipDefaultCheckout()
           }
-          steps {
+          steps{
             deleteDir()
-            dir("${OPBEANS_REPO}"){
-              git credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
-                  url: "git@github.com:elastic/${OPBEANS_REPO}.git"
-              // It's required to transform the tag value to the artifact version
-              sh script: ".ci/bump-version.sh ${env.BRANCH_NAME.replaceAll('^v', '')}", label: 'Bump version'
-              // The opbeans pipeline will trigger a release for the master branch
-              gitPush()
-              // The opbeans pipeline will trigger a release for the release tag
-              gitCreateTag(tag: "${env.BRANCH_NAME}")
+            unstash 'source'
+            // Bail out if we don't have the right parameters
+            isValidReleaseVer("${params.release_ver}")
+            // Prepare our release toolchain
+            sh(script: ".ci/scripts/prepare-release-containers.sh")
+            // Prepare GitHub environment
+            githubEnv()
+            // Generate a preview of the CHANGELOG file for visual inspection
+            script{
+              def credentialsId = params.get('credentialsId', '2a9602aa-ab9f-4e52-baf3-b71ca88469c7')
+              withCredentials([string(credentialsId: "${credentialsId}", variable: 'GITHUB_TOKEN')]) {
+                def preview = sh(script: "./.ci/scripts/generate_changelog.sh --token ${GITHUB_TOKEN} --version ${params.release_ver} --preview true", returnStdout: true)
+              }
+              input(message: 'Do you accept these proposed entries for the CHANGELOG.asciidoc file?', ok: 'Yes, proceed with writing them.',
+                    parameters: [text(description: 'IMPORTANT: Proposed changes CANNOT be edited here. Abort the build if not acceptable.',
+                    defaultValue: "${preview}", name: 'preview')])
+              withCredentials([string(credentialsId: "${credentialsId}", variable: 'GITHUB_TOKEN')]) {
+                // Re-run without preview flag and rewrite the file
+                sh(script: "./.ci/scripts/generate_changelog.sh --token ${GITHUB_TOKEN} --version ${params.release_ver} --preview false > ./CHANGELOG.asciidoc", returnStdout: true)
+              }
+            }
+            // Update the EOL table in the upgrading.asciidoc file
+            sh(script: "./.ci/scripts/generate_eol.sh --version ${params.release_ver} > ./docs/upgrading.asciidoc")
+            // Update package.json
+            sh ".ci/scripts/update_package_version.sh ${params.release_ver}"
+            // Tag and push
+            sh (script: "./.ci/scripts/tag_release.sh --version ${params.release_ver}")
```

NIT and suggestion:
- add docs to the label, 
- if exec privileges in the script then it's not required `./`

```suggestion
            sh(label: 'Tag and push', script: ".ci/scripts/tag_release.sh --version ${params.release_ver}")
```